app/models/data_resource.rb

```diff
@@ -10,4 +10,14 @@ class DataResource < ApplicationRecord
 
   enum :access_rights, { INTERNAL: 0, OPEN: 1, COMMERCIAL: 2 }
   enum :security_classification, { OFFICIAL: 0, SECRET: 1, TOP_SECRET: 2 }
+
+  include PgSearch::Model
+  pg_search_scope :search,
+                  against: { title: 'A', description: 'B', summary: 'C' },
+                  associated_against: { publisher: { name: 'D' } },
+                  using: { tsearch: { prefix: true, dictionary: 'english' } }
+
+  def summary
+    read_attribute(:summary).presence || self[:description]&.truncate(250)
```

Comment:
You can do something a little neater here:
```suggestion
    read_attribute(:summary).presence || description&.truncate(250)
```
You would use `self[:description]` (or `self.description` which is more or less the same) if you wanted to set the value of `description`, but here we just want to read it so we can just get it with `description`.